internal/sudo/contract_manager.go

```diff
@@ -75,37 +74,72 @@ type OpenAckDetails struct {
 	CounterpartyVersion   string `json:"counterparty_version"`
 }
 
+type ContractMethods interface {
+	HasContractInfo(sdk.Context, sdk.AccAddress) bool
+	Sudo(sdk.Context, sdk.AccAddress, []byte) ([]byte, error)
+}
+
+var _ ContractMethods = (*ContractManager)(nil)
+
+type ContractManager struct {
+	wasmKeeper ContractMethods
+}
+
 type Handler struct {
-	moduleName string
-	wasmKeeper *wasm.Keeper
+	moduleName      string
+	contractManager *ContractManager
 }
 
-func (s *Handler) Logger(ctx sdk.Context) log.Logger {
-	return ctx.Logger().With("module", fmt.Sprintf("x/%s", s.moduleName))
+func NewContractManager() ContractManager {
+	return ContractManager{}
 }
 
-func NewSudoHandler(wasmKeeper *wasm.Keeper, moduleName string) Handler {
+func (cm *ContractManager) NewSudoHandler(moduleName string) Handler {
 	return Handler{
-		moduleName: moduleName,
-		wasmKeeper: wasmKeeper,
+		moduleName:      moduleName,
+		contractManager: cm,
 	}
 }
 
-func (s *Handler) SudoResponse(
+func (cm *ContractManager) SetWasmKeeper(wasmKeeper ContractMethods) {
+	cm.wasmKeeper = wasmKeeper
+}
+
+func (cm *ContractManager) HasContractInfo(ctx sdk.Context, contractAddress sdk.AccAddress) bool {
+	if cm.wasmKeeper == nil {
+		panic("wasmKeeper pointer is nil")
+	}
```

Comment:
What's the point of this check and panic? Without this check, you will get a panic on nil pointer dereference anyway.

Reply:
Simply to have more meaningful error.

Reply:
eh, I guess it will be meaningful enough anyway without panicing manually. You are free to close this thread without deleting this code though. I will not open this thread again.